plugins/lsp/conform-nvim.nix

```diff
@@ -1,33 +1,101 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
-with lib;
 let
-  cfg = config.plugins.conform-nvim;
+  inherit (lib) types;
+  inherit (lib.nixvim) defaultNullOpts;
 in
-{
-  options.plugins.conform-nvim = helpers.neovim-plugin.extraOptionsOptions // {
-    enable = mkEnableOption "conform-nvim";
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "conform-nvim";
+  originalName = "conform.nvim";
+  defaultPackage = pkgs.vimPlugins.conform-nvim;
+
+  maintainers = [ lib.maintainers.khaneliman ];
+
+  # TODO: added 2024-08-23 remove after 24.11
+  deprecateExtraOptions = true;
+  optionsRenamedToSettings = [
+    "formatters"
+    "formattersByFt"
+    "logLevel"
+    "notifyOnError"
+  ];
+  imports =
+    map
+      (
+        optionName:
+        lib.mkRemovedOptionModule [
+          "plugins"
+          "conform-nvim"
+          optionName
+        ] "Please use `plugins.conform-nvim.settings.${lib.nixvim.toSnakeCase optionName}` instead."
```

Comment:
This should ideally also note that sub-option(s) (if any) would also be snake_case.

Unfortunately there's no (simple[^1]) way to know if the user defined a string or an attrset, so the error message would have to be somewhat generic.

[^1]: Accessing `config.*` would throw, as would `option.*.value`. We could merge `option.*definitions` ourselves, but it's not worth it just to provide a better error message!

Reply:
This look better?